pkg/workload/workload.go

```diff
@@ -857,6 +857,20 @@ func ApplyAdmissionStatusPatch(ctx context.Context, c client.Client, patch *kueu
 	return c.Status().Patch(ctx, patch, client.Apply, client.FieldOwner(constants.AdmissionName), client.ForceOwnership)
 }
 
+// PatchAdmissionStatus updates the admission status of a workload.
+// If the WorkloadRequestUseMergePatch feature is enabled, it uses a Merge Patch with update function.
+// Otherwise, it runs the update function and, if updated, applies the SSA Patch status.
+func PatchAdmissionStatus(ctx context.Context, c client.Client, wOrig *kueue.Workload, w *kueue.Workload, strict bool, clk clock.Clock, update func() (bool, error)) error {
```

Comment:
This API seems very weird. I think you should only need to accept wOrig  + update.